### PR TITLE
Pthread stub

### PR DESCRIPTION
**Pull request by roblabla**
So this is a cut down version of the pthread PR with *everything* stubbed. Meaning nothing actually works, it all returns ENOSYS.

This will allow dvdfreitag to easily work on his C++ support work.

Should the commit history be squashed before merging ?

### CODE REVIEW

**Comment by roblabla** [comment]
@misson20000 I'm thinking of merging this so @dvdfreitag can go forward with C++ support, and rebasing my actual pthread implementation on top of this. This way both works can go forward independently. Are you OK with this approach ?

File: include/libtransistor/tls.h [hunk 1]
```diff
@@ -5,7 +5,20 @@
...
+#include <reent.h>
+
+struct thread_ctx {
+	struct _reent reent;
+	void *pthread; // Pointer to pthread internal structure
+};
+
+struct tls {
+	char ipc_buffer[0x100];
+	char _unk1[0xF8];
```
Comment (misson20000) [inline include/libtransistor/tls.h]: would feel more comfortable with these as `uint8_t` arrays

File: pthread/Makefile [hunk 2]
```diff
@@ -0,0 +1,50 @@
+#	$OpenBSD: Makefile,v 1.49 2017/10/15 23:40:33 guenther Exp $
```
Comment (misson20000) [inline pthread/Makefile]: This Makefile looks trivial. Can we avoid using recursive make and just integrate this into the main Makefile?
While we're at it, put object files in build/ and disable some warnings for pthreads?

File: Makefile [hunk 3]
```diff
@@ -84,6 +87,11 @@ $(LIBTRANSISTOR_HOME)/build/newlib/Makefile:
 $(LIBTRANSISTOR_HOME)/build/newlib/aarch64-none-switch/newlib/libc.a: $(LIBTRANSISTOR_HOME)/build/newlib/Makefile
 	$(MAKE) -C $(LIBTRANSISTOR_HOME)/build/newlib/
 
+$(LIBTRANSISTOR_HOME)/pthread/libpthread.a:
+	$(MAKE) -C $(LIBTRANSISTOR_HOME)/pthread
+
+.PHONY: $(LIBTRANSISTOR_HOME)/pthread/libpthread.a
+
```
Comment (misson20000) [inline Makefile]: would prefer to avoid the use of recursive make. see comments on pthread/Makefile

File: pthread/Makefile (hunk 2 above)
Comment (misson20000) [reply pthread/Makefile]: In order to keep the Makefile slim we might want to put rules relating to pthreads in a separate Makefile that we include from the main one. I just don't want to be using recursive make if we don't have to.

**Comment by roblabla** [comment]
Is this fine ?

**Pull request merged**